catalyst-gateway/bin/src/db/event/signed_docs/signed_doc_body.rs

```diff
@@ -0,0 +1,107 @@
+//! `SignedDocBody` struct implementation.
+
+use super::DocsQueryFilter;
+use crate::{
+    db::event::{common::query_limits::QueryLimits, EventDB},
+    jinja::{get_template, JinjaTemplateSource},
+};
+
+/// Filtered select sql query jinja template
+pub(crate) const FILTERED_SELECT_SIGNED_DOCS_TEMPLATE: JinjaTemplateSource = JinjaTemplateSource {
+    name: "filtered_select_signed_documents.jinja.template",
+    source: include_str!("./sql/filtered_select_signed_documents.sql.jinja"),
+};
+
+/// Signed doc body event db struct
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct SignedDocBody {
+    /// `signed_doc` table `id` field
+    id: uuid::Uuid,
+    /// `signed_doc` table `ver` field
+    ver: uuid::Uuid,
+    /// `signed_doc` table `type` field
+    doc_type: uuid::Uuid,
+    /// `signed_doc` table `author` field
+    author: String,
+    /// `signed_doc` table `metadata` field
+    metadata: Option<serde_json::Value>,
+}
+
+impl SignedDocBody {
+    /// Returns the document id.
+    pub(crate) fn id(&self) -> &uuid::Uuid {
+        &self.id
+    }
+
+    /// Returns the document version.
+    pub(crate) fn ver(&self) -> &uuid::Uuid {
+        &self.ver
+    }
+
+    /// Returns the document author.
+    pub(crate) fn author(&self) -> &String {
+        &self.author
+    }
+
+    /// Returns all signed document fields for the event db queries
+    pub(crate) fn postgres_db_fields(&self) -> [&(dyn tokio_postgres::types::ToSql + Sync); 5] {
+        [
+            &self.id,
+            &self.ver,
+            &self.doc_type,
+            &self.author,
+            &self.metadata,
+        ]
+    }
+
+    /// Creates a  `SignedDocBody` instance.
+    #[allow(dead_code)]
+    pub(crate) fn new(
+        id: uuid::Uuid, ver: uuid::Uuid, doc_type: uuid::Uuid, author: String,
+        metadata: Option<serde_json::Value>,
+    ) -> Self {
+        Self {
+            id,
+            ver,
+            doc_type,
+            author,
+            metadata,
+        }
+    }
+
+    /// Loads a vector of `SignedDocBody` from the event db.
+    #[allow(dead_code)]
+    pub(crate) async fn load_from_db(
+        conditions: &DocsQueryFilter, query_limits: &QueryLimits,
+    ) -> anyhow::Result<Vec<Self>> {
```

Comment:
Return an async iterator that iterates over the results as the data set could be very large.
Do not query the whole result set and then return an iterator over that.
Use [`query_raw`](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.Client.html#method.query_raw)
And return an async iterator which wraps the [`RowStream`](https://docs.rs/tokio-postgres/latest/tokio_postgres/struct.RowStream.html) it returns